spark/src/main/scala/org/apache/comet/CometExecIterator.scala

```diff
@@ -133,23 +136,51 @@ class CometExecIterator(
     }
   }
 
-  def getNextBatch(): Option[ColumnarBatch] = {
+  private def getNextBatch: Option[ColumnarBatch] = {
     assert(partitionIndex >= 0 && partitionIndex < numParts)
 
     if (tracingEnabled) {
       traceMemoryUsage()
     }
 
     val ctx = TaskContext.get()
-    withTrace(
-      s"getNextBatch[JVM] stage=${ctx.stageId()}",
-      tracingEnabled, {
-        nativeUtil.getNextBatch(
-          numOutputCols,
-          (arrayAddrs, schemaAddrs) => {
-            nativeLib.executePlan(ctx.stageId(), partitionIndex, plan, arrayAddrs, schemaAddrs)
-          })
-      })
+
+    try {
```

Comment:
This will work for the time being, but perhaps we should trap the error in `errors.rs`?

Reply:
We'l have to think about that design since every error handler could necessitate a different set of configs/context to be sent over for correct behavior on the native side.

Reply:
Maybe log an issue to see if the error/exception translation can be improved?